sql-plugin/src/main/301until330-all/scala/com/nvidia/spark/rapids/shims/v2/Spark30Xuntil32XShims.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.nvidia.spark.rapids.shims.v2
+
+import com.nvidia.spark.rapids._
+
+import org.apache.spark.sql.catalyst.csv._
+import org.apache.spark.sql.execution._
+import org.apache.spark.sql.execution.datasources.v2._
+
+
+trait Spark30Xuntil32XShims extends SparkShims {
```

Comment:
this naming seems odd we have  Spark30Xuntil32XShims in a 301until330-all directory.   We don't need this for 33x since it looks like you separated it out from Spark32xShim right? if that is the case the 301until330-all directory I think can go away.

Reply:
the point of this directory to contribute  common code to all shims from 3.0.x to 3.2.x excluding 3.3.x

Reply:
ah right, my mistake, but shouldn't this class be called Spark30Xuntil330Shims?  until32x implies that it doesn't include 32X version which is confusing because you are using it:

>class Spark321Shims extends Spark32XShims with Spark30Xuntil32XShims

Reply:
yes, you are right it was supposed to be consistent with the directory. 

The general idea I had is to start consolidating duplicated code in these kind of version range classes.